File: src/main/java/social/alone/server/event/repository/EventRepositoryCustom.java
package social.alone.server.event.repository;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import social.alone.server.event.Event;
import social.alone.server.event.EventQueryParams;
import social.alone.server.user.User;

public interface EventRepositoryCustom {
  Page<Event> search(Pageable pageable, User user, EventQueryParams eventQueryParams);
  Page<Event> search(Pageable pageable, EventQueryParams eventQueryParams);
}

File: src/main/kotlin/social/alone/server/event/repository/EventRepositoryCustom.kt
package social.alone.server.event.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import social.alone.server.event.domain.Event
import social.alone.server.event.type.EventQueryParams

interface EventRepositoryCustom {
    fun search(pageable: Pageable, eventQueryParams: EventQueryParams): Page<Event>
}
